### DISCUSSION TITLE
Contributing

### DISCUSSION THREAD

**Discussion by mrspence** [ChartsCSS/charts.css#21] [General]
Hey 👋

I've spent the last few months tinkering with your framework, I wonder if you'd be interested in code contributions. I really like the direction you've taken and would love to help. I'm a full stack developer working full time, but I do see big potential in this framework and would love to spare some time in growing it.

Leaves me with a few questions:

- As the author, would you be interested in contributions via pull requests at this point? 

Assuming so,

- Where would you want contributions at the moment?

**Comment by mrspence** [comment]
To add - I'm developing a [Vue JS wrapper for this framework](https://github.com/Vue-Charts-CSS/vue.charts.css) so it can be used interactively within Vue projects. Originally I was writing a private wrapper, but realised it may be useful for others too.

```vue
<template>
<charts-css
        type="bar"
        heading="Team's coffee count"
        :labels="coffeeLabels"
        :datasets="coffeeDatasets"
    />
</template>

<script>
    export default {
        data()
        {
            return {
                coffeeLabels: [ "Monday", "Tuesday", "Wednesday", "Thursday", ],
                coffeeDatasets: [
                    {
                        name: "Ben",
                        values: [1, 3, 2, 12],
                    },
                    {
                        name: "Tim",
                        values: [1, 3, 1, 2],
                    },
                    {
                        name: "John",
                        values: [72, 227, 0, 0],
                    },
                ],
            };
        }
    }
</script>
```

This wrapper is geared towards making it easier to pass and render data, but still retains the benefits of this CSS framework: it encourages the user to style with CSS Selectors, control the layout of charts and retain valuable accessibility.

The Vue wrapper is more suited for when the developer needs to dynamically change data or make charts interactive. This is without messy procedural JavaScript (or the horrifying prospect of jQuery).

Of course a developer can also use the CSS framework directly in Vue projects too, though drawing from my own experience this could get quite heavy to code the template (html) when different datasets are being dynamically rendered for a chart.

With this wrapper, it makes it easy to dynamically change:

- Chart type
- Datasets
- Classes (via props or directly through a `classes` prop)
- Rendering of values, labels, legends and data
- Heading
- and more

At the moment it is still very much under development, but includes initial support for these charts: Bar, Column, Line, Area 👍

There's more to do yet, following the [public roadmap here](https://vue-charts-css.github.io/docs/development/roadmap/) _(subject to change)_.

Anyway, I'd really like to see **Charts.CSS** grow and I'm open to contributing if you'd like more hands. 👍

**Reply by ramiy** [reply]
@mrspence  Awesome work! 

Currently I'm working on the next version. I'll push the latest commits to GitHub and update the [Roadmap](https://chartscss.org/development/roadmap/).  This way you can follow the work and contribute PRs .

As for the Vue wrapper, love it!

I purchased two domains [ChartsCSS.org](https://chartscss.org/) and [ChartsCSS.com](https://chartscss.com/). The first is for the core framework and the second for addons, plugins and themes. We can publish your **Vue JS wrapper for ChartsCSS** at ChartsCSS.com.

**Reply by mrspence** [reply]
Thanks - and that's brilliant! Is your next draft version on a dev branch?

And thanks again!

Brilliant, I did discover your addons/plugins website, it's a smashing idea separating it all out! Keeps the framework super lean and hopefully encourages more developers to contribute 👍 Anyway, I'll certainly get the wrapper up to a release version so it deserves a spot.
